src/levanter/eval_harness.py

```diff
@@ -0,0 +1,564 @@
+        result = result[:initial_length]
+
+        return result
+
+    def loglikelihood_rolling(self, requests) -> List[Tuple[float]]:
+        raise NotImplementedError()
+
+    def generate_until(self, requests) -> List[str]:
+        raise NotImplementedError()
+
+
+@functools.partial(jax.jit, static_argnums=(0, 3))
+def _jit_create_example(Pos, tokens, prompt_len, pad_token_id):
+    tokens = hax.named(tokens, Pos)
+    return LmExample.from_prompt_and_completion(Pos, tokens, prompt_len, ignore_id=pad_token_id)
+
+
+@dataclass(frozen=True)
+class TaskConfig:
+    """
+    This is a dataclass that represents the configuration for a task in the LM Eval Harness. It is used to specify
+    the configuration for a task in the LM Eval Harness, and is used to generate the task dictionary that the LM Eval
+    Harness expects.
+
+    nb that LM Eval Harness has its own TaskConfig, but its defaults are not the same as just passing in
+    a dict, and we want the behavior of passing in a dict.
+
+    See Also:
+        [LM Eval Harness TaskConfig](https://github.com/EleutherAI/lm-evaluation-harness/blob/0ef7548d7c3f01108e7c12900a5e5eb4b4a668f7/lm_eval/api/task.py#L55)
+    """
+
+    task: str
+    task_alias: str | None = None
+    num_fewshot: int | None = None
+
+    use_prompt: str | None = None
+    description: str | None = None
+    target_delimiter: str | None = None
+    fewshot_delimiter: str | None = None
+
+    def to_dict(self):
+        base_dict = dataclasses.asdict(self)
+        return {k: v for k, v in base_dict.items() if v is not None}
+
+
+@dataclass(frozen=True)
+class LmEvalHarnessConfig:
+    task_spec: list[TaskConfig | str] | None = None
+    max_examples: int | None = None
+    max_eval_length: int | None = None
+    log_samples: bool = False
+
+    def task_spec_or_default(self) -> list[str | dict]:
+        if self.task_spec is None:
+            return ["hellaswag", "piqa"]
+        return [task.to_dict() if isinstance(task, TaskConfig) else task for task in self.task_spec]
+
+    def to_task_dict(self) -> dict:
+        import lm_eval.tasks as tasks
+
+        manager = tasks.TaskManager()
+        # we need to do it this way b/c i can't figure out how to run e.g. hellaswag 0 shot and 10 shot in a single run
+        this_tasks = {}
+        for task in self.task_spec_or_default():
+            try:
+                if isinstance(task, str):
+                    this_tasks.update(tasks.get_task_dict(task, manager))
+                else:
+                    our_name = task.get("task_alias", task["task"]) if isinstance(task, dict) else task
+                    our_name = our_name.replace(" ", "_")
+                    task_dict = tasks.get_task_dict([task], manager)
+                    this_task = task_dict.popitem()[1]
+                    # hacky, but this allows us to run multiple instances of the same task with different fewshot settings
+                    this_task.config.task = our_name
```

Comment:
I think this works well since the result file is saved as `task.jsonl` so using the alias helps us distinguish the number of shots. It might be worth it to move this logic into a helper function `create_task_with_unique_name()` for readability...

Reply:
good idea!